### PR TITLE
Salocinx master

### PR DESCRIPTION
**Pull request by z3t0**
#413 

@salocinx

I needed to recreate the PR to add the changelog commit. I will give the code another look in the next few days and then merge within the week.

Thank you very much! Your contribution is greatly appreciated. If possible, could you summarize in a sentence or two any difficulties you may have had in creating this protocol spec? Just something to add to the wiki for any future coders that want to add their own protocols.

### CODE REVIEW

**Comment by z3t0** [comment]
The branch has become out of date again, should be a trivial fix. Also the new protocol needs to be added to the relevant example sketches. And we need documentation to explain the protocol specs

**Comment by ArminJo** [comment]
Please rebase the code if the protocol is still required.
